src/component-library/LoadingSpinner/LoadingSpinner.tsx

```diff
@@ -0,0 +1,10 @@
+import { BaseLoadingSpinner, BaseLoadingSpinnerProps } from './LoadingSpinner.style';
+
+type LoadingSpinnerProps = BaseLoadingSpinnerProps & React.ComponentPropsWithRef<'span'>;
```

Comment:
Aren't you missing the forwardRef on this component, since you are using `ComponentPropsWithRef` interface

Reply:
When necessary, I think we can add `forwardRef`. For now, I do not see any necessities for that.